web-masons/pyramid-zappa-api-boilerplate
# Ansible vagrant init

## Overview
I have added Vagrant and Ansible to the project.

Resolves #8

### Demo

Optional. Screenshots, `curl` examples, etc.

### Notes
The Vagrant file calls out to the script/bootstrap file for provisioning.

The bootstrap script then uses apt to update and make sure all Ansible requirements exist on the box. This is hopefully a temporary solution. I would like to create a Python Development box so that all the ansible/infrastructure stuff related to generic “python development” is covered by Packer generating a box.

From there, the bootstrap script calls out to the requirements.yml for any Ansible Galaxy roles we might need. It then calls the environment specific playbook (in this case vagrant.yml).

Once all that is done, we’ve got the basics of a Python dev environment built up with AWS, BOTO, etc.

## Testing Instructions

 * Make sure you have the project set up according to the README
 * Run `vagrant status` to make sure you don’t have an environment already running. If you do, `vagrant destroy` it.
 * Run `vagrant up`, the script should execute properly
 * Run `vagrant ssh`
 * Run “aws help” and you should receive the appropriate prompts from the AWS command line, meaning the install worked correctly.
